### PR TITLE
Jason/fix 384

### LINKED ISSUE #384: regression: toolbar drag sources are broken
The toolbar's drag sources seem to have been broken by a recent commit or package upgrade. This means that new widgets can no longer be docked, but must first be created as floating widgets, and then docked.

### PR DESCRIPTION
**Pull request by duidae**
close #384 

dragsource should be created after golden layout object is instantiated, originally created in componentDidUpdate() in ToolbarMenuComponent will not be effective in the very 1st load(golden layout object is instantiated in componentDidMount() in App.tsx, which is later than componentDidUpdate() of ToolbarMenuComponent according to react life cycle)

### CODE REVIEW

File: src/stores/WidgetsStore.ts [hunk 1]
```diff
@@ -140,7 +152,22 @@ export class WidgetsStore {
         }
     };
 
-    // region Golden Layout Widgets
+    // create drag source for ToolbarMenuComponent
```
Comment (veggiesaurus) [inline src/stores/WidgetsStore.ts]: can this be made `private static`?

Comment (duidae) [reply src/stores/WidgetsStore.ts]: cant be static because the need of appStore instance for props in glConfig. is there a better way for this?

Comment (duidae) [reply src/stores/WidgetsStore.ts]: ah how about I add a new parameter of appStore

Comment (duidae) [reply src/stores/WidgetsStore.ts]: modification added

**Comment by duidae** [comment]
add check to image viewer to for 0 width/height prevention, should work as following:

out-1 ogv

**Comment by veggiesaurus** [comment]
I'm not sure that these changes are right. You've solved #384 , but the changes to the other stuff seems unrelated. Can isolate the fixes to #384 in this branch and discuss the other changes in another PR?

I'd like to keep rendering the `RasterViewComponent` at all times, even if it has zero width and height, since this prevents the WebGL context from being lost, and thus the GPU cache from being cleared. 

I'm also not entirely sure about adjusting the zoom when resizing the view. If I zoom in to 1x, and then resize the view, it jumps back to my auto-scaled zoom level. This would only be considered correct if I was currently in "zoom to fit" zoom. I'd prefer us to respect the user's zoom level choice.

**Comment by duidae** [comment]
reverted. I think we need to add error handling for 0 width/height to RasterViewComponent/OverlayComponent/CursorOverlayComponent/BeamProfileOverlayComponent/RegionViewComponent/, 

since dragsource may stack image viewer to not-major tab, and dynamically changing layout also resize each area to 0 and then re-layout

**Comment by duidae** [comment]
the rest is in  PR #406

**Pull request merged**